contracts/SL2RD.sol

```diff
@@ -164,8 +242,40 @@ contract SL2RD is
 
         // Call the parent's _transfer function
         super.transferFrom(from_, to_, tokenId_);
+
+        // Update the transfer timestamp
+        _latestTransferTimestamps[tokenId_] = block.timestamp;
+    }
+
+    /// @notice Returns the latest transfer timestamp for each given tokenId.
+    function splitTransferTimestamps(
+        uint256[] memory tokenIds
+    ) public view returns (uint256[] memory) {
+        uint256 length = tokenIds.length;
+        uint256[] memory timestamps = new uint256[](length);
+        for (uint256 i = 0; i < length; i++) {
+            timestamps[i] = _latestTransferTimestamps[tokenIds[i]];
+        }
+        return timestamps;
     }
 
+    // /// @notice Function to transfer the next available slot to a specified address.
+    // /// This function keeps track of the available community splits and the next available slot.
+    // /// It reverts if all community apportioned splits are allocated.
+    // function transferNextAvailable(
```

Comment:
@DeusExM4ch1n4 Here we need to ensure that the distribution is probabilistically fair, eg. in S2RD we use the initialization logic to ensure a random distribution. Is the slot ordering inherited here, or does the creator essentially get allocated the first _N_ slots as part of their reservation such that the first _N_ transactions go to the creator?